stan/math/rev/core/arena_allocator.hpp

```diff
@@ -85,7 +85,7 @@ struct arena_allocator {
    * No-op. If anything needs destroyed on the stack this happens
    *  when calling `recover_memory()`.
    */
-  static inline void destroy(pointer p) { (static_cast<T*>(p))->~T(); }
+  static inline void destroy(pointer p) { }
```

Comment:
@t4c1 should this actually call the destructor or no? Running with clang sanitizer it doesn't throw errors just having this be empty which makes sense to me since the dynamic mem is all allocated on the stack allocator

Reply:
I don't think we need that fucntion. See https://github.com/stan-dev/math/pull/2418#discussion_r594128710